odins_spear/scripts/locate_free_extension.py

```
@@ -0,0 +1,47 @@
+from exceptions import OSExtensionNotFound
+
+def retrieve_extensions( api, service_provider_id: str, group_id: str ) -> list:
+    
+    extensions = []
+
+    dataset = (
+       api.get.users( service_provider_id, group_id )              +
+       api.get.group_hunt_groups( service_provider_id, group_id )  +
+       api.get.group_call_centers( service_provider_id, group_id ) +
+       api.get.auto_attendants( service_provider_id, group_id )
+    )
+
+    for data in dataset:
+        if not data['extension']:
+            continue
+
+        extensions.append(int(data['extension']))
+
+    return extensions if extensions else None
+
+def main( api, service_provider_id: str, group_id: str, range_start: int, range_end: int ):
+    '''Retrieves The Lowest Free Extension Available In The Designated Group Passed.
+    '''
+
+    if range_start > range_end:
```

This should raise an error dude rather than correcting for the user. 

Python Zen - 'Explicit is better than implicit' 

Python Zen is like the guide lines for writing Python code: https://peps.python.org/pep-0020/